### PR TITLE
GLFW tests

### PR DESCRIPTION
**Pull request by HurricanKai**
Adds some basic GLFW tests. These don't exactly provide good coverage or anything, but they are tests and will catch (and have already caught) some bugs.

### CODE REVIEW

File: Silk.NET.sln [hunk 1]
```diff
@@ -357,6 +357,10 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Silk.NET.Maths.Tests", "src
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Utf8Test", "src\Lab\Utf8Test\Utf8Test.csproj", "{C06556DC-52AF-44AE-99B8-1251C18C3973}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Silk.NET.GLFW.Tests", "src\Windowing\Silk.NET.GLFW.Tests\Silk.NET.GLFW.Tests.csproj", "{043D78F1-3D65-4439-BDEB-EBA10DE44810}"
```
Comment (Perksey) [inline Silk.NET.sln:360]: I think we should move all of the tests into a new `tests/` directory. That way they're easily accessible.

Comment (HurricanKai) [reply Silk.NET.sln:360]: Good point. Would you also move the maths tests there, or just the other(s)?

Comment (Perksey) [reply Silk.NET.sln:360]: Probably all of them.

**Comment by HurricanKai** [comment]
In regards to the CI failure - note that these tests are currently failing due to Ultz/NativePackages#1

File: Silk.NET.sln (hunk 1 above)
Comment (HurricanKai) [reply Silk.NET.sln:360]: In that case I'd prefer to make another PR moving them, just for that sweet git log 😄 
Would that be ok with you?

Comment (Perksey) [reply Silk.NET.sln:360]: Yeah sure. Will move onto the actual review after a dog walk.

File: src/Lab/GammaFun/Program.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                glfw.Terminate();
+                return;
+            }
+
+            var ramp = glfw.GetGammaRamp(monitor);
+            var size = (int)(ramp->Size * sizeof(short));
+
+            var red = (short*)SilkMarshal.Allocate(size);
+            var green = (short*)SilkMarshal.Allocate(size);
+            var blue = (short*)SilkMarshal.Allocate(size);
```
Comment (Perksey) [inline src/Lab/GammaFun/Program.cs:30]: Afaik GLFW copies everything once you send the ramp over, so these can be stackallocs.

File: src/Windowing/Silk.NET.GLFW.Tests/GammaTests.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                glfw.Terminate();
+                throw new XunitException("Failed to create window");
+            }
+
+            var ramp = glfw.GetGammaRamp(monitor);
+            var size = (int)(ramp->Size * sizeof(short));
+
+            var red = (short*)SilkMarshal.Allocate(size);
+            var green = (short*)SilkMarshal.Allocate(size);
+            var blue = (short*)SilkMarshal.Allocate(size);
```
Comment (Perksey) [inline src/Windowing/Silk.NET.GLFW.Tests/GammaTests.cs:38]: Same again

File: src/Windowing/Silk.NET.GLFW.Tests/TitleTests.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+//
+// 2. Altered source versions must be plainly marked as such, and must not
+//    be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source
+//    distribution.
+//
+//========================================================================
+
+// ported from https://github.com/glfw/glfw/blob/master/tests/title.c
```
Comment (Perksey) [inline src/Windowing/Silk.NET.GLFW.Tests/TitleTests.cs:31]: Should go before the license header of title.c

**Comment by Perksey** [comment]
Any plans to rectify the arguably small fixes requested?

File: src/Lab/GammaFun/Program.cs (hunk 2 above)
Comment (HurricanKai) [reply src/Lab/GammaFun/Program.cs:30]: But isn't that an unbound stackalloc and may cause a stackoverflow?

**Comment by HurricanKai** [comment]
Sorry, lost track of this.

File: src/Lab/GammaFun/Program.cs (hunk 2 above)
Comment (Perksey) [reply src/Lab/GammaFun/Program.cs:30]: Technically yes, but in practice this is usually a pretty small number. I don't ever see this going above 256 bytes allocated, to be honest. But your call.

Comment (HurricanKai) [reply src/Lab/GammaFun/Program.cs:30]: Fair enough, will change soon

File: src/Windowing/Silk.NET.GLFW.Tests/ClipboardTests.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
+﻿// This file is part of Silk.NET.
```
Comment (Perksey) [inline src/Windowing/Silk.NET.GLFW.Tests/ClipboardTests.cs:1]: Update license headers and merge main, also.

**Review by Perksey — CHANGES_REQUESTED** [review]

**Comment by Perksey** [comment]
Is this dead?

**Comment by Perksey** [comment]
Going to close this as it hasn't seen a commit since December, the PR history is here if ever this work continues.

**Pull request closed without merging**